### DISCUSSION TITLE
Add enum with http status codes

### DISCUSSION THREAD

**Discussion by christophstockinger** [laravel/framework#47333] [Ideas] [answered]
I personally write APIs with Laravel very often, so I also need the HTTP status codes very often.
When programming the APIs, I have always noticed myself that I can speak the statis (for example: "Not Found") by heart, but now and then I can't think of the right number for it.
For this reason I thought about whether it would make sense for Laravel to provide an enum in the core (also against the background that PHP 8.1 now supports enums), which provides all HTTP status codes.

I have already prepared a suitable branch for this in my fork: https://github.com/christophstockinger/laravel-framework/tree/feat/enum-http-status-codes

I also add a PR #47334 

An example for the usage would be:
```php
public function index(Request $request) {
     // ... some databse logic ... 
    return response()->json(['some data'], HttpStatus::OK)
}

public function store($data, Request $request) {
    // ... some databse logic ... 
    return response()->json(['created data'], HttpStatus::CREATED)
}
```

I look forward to feedback on the feature request as well as feedback on the code.

**Comment by henzeb** [comment]
It should be `HttpCode::Ok` or `HttpCode:;Created`.

Also, there are constants you can use in the Symfony Response class

**Comment by EhsanTava** [comment]
Henzeb is right , you can use Symfony Response's consts .

Symfony\Component\HttpFoundation\Response

All statuses are defined there .

**Reply by grafst** [reply, marked as answer]
Here the link, for convenience: https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpFoundation/Response.php
